### PR TITLE
fixes #13218, #13239 - make job template name unique, no longer require category

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: db/migrate/20160125155108_make_job_template_name_unique.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    duplicates = JobTemplate.unscoped.group(:name).count.delete_if { |_name, value| value == 1 }
+    duplicates.each_key do |duplicated_name|
+      templates = JobTemplate.where(:name => duplicated_name)
+      templates.each_with_index do |template, index|
+        new_name = JobTemplate.exists?(:name => "#{template.name}-#{index}") ? "#{template.name}-#{index}-#{SecureRandom.hex(2)}" : "#{template.name}-#{index}"
+        template.update_attribute(:name, new_name)
+      end
+    end
+  end
+end
```
Comment (stbenjam) [inline db/migrate/20160125155108_make_job_template_name_unique.rb]: Taken from https://github.com/theforeman/foreman/blob/develop/db/migrate/20141021105446_rename_subnet_name_to_unique.rb

**Comment by stbenjam** [comment]
Still a WIP, a few issues by changing how this works

**Comment by stbenjam** [comment]
Should be good to go, hammer README needs updating too -> https://github.com/theforeman/hammer_cli_foreman_remote_execution/issues/9

**Comment by adamruzicka** [comment]
ACK

**Pull request merged**